.circleci/circle-ci-config.py

```diff
@@ -0,0 +1,196 @@
+        'MIDRES': {
+            # apply defaults to all jobs, so the below just overrides jobs which do not match the default
+            '_default_job_':                        {'parallelism': 50,    'exec_resource_class': 'large'},
+            'jobs': {
+                # large 10
+                'j8_jvm_dtests':                    {'parallelism': 10},
+                'j8_jvm_dtests_vnode':              {'parallelism': 10},
+                # large 4
+                'j8_jvm_upgrade_dtests':            {'parallelism': 4},
+                # medium 10
+                'j11_jvm_dtests':                   {'parallelism': 10, 'exec_resource_class': 'medium'},
+                'j11_jvm_dtests_vnode':                   {'parallelism': 10, 'exec_resource_class': 'medium'},
+
+                # xlarge 100
+                'j8_upgradetests-no-vnodes':        {'parallelism': 100, 'exec_resource_class': 'xlarge'},
+
+                # xlarge 25
+                'repeated_upgrade_dtest':           {'parallelism': 25, 'exec_resource_class': 'xlarge'},
+
+                # large 25
+                'j11_repeated_dtest':               {'parallelism': 25},
+                'j8_repeated_dtest':                {'parallelism': 25},
+                'repeated_jvm_upgrade_dtest':       {'parallelism': 25},
+
+                # medium 25
+                'j11_repeated_utest':               {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'j11_unit_tests':                   {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'j8_repeated_utest':                {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'j8_unit_tests':                    {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'utests_compression':               {'parallelism': 25, 'exec_resource_class': 'medium'},
+                'utests_system_keyspace_directory': {'parallelism': 25, 'exec_resource_class': 'medium'},
+
+                'utests_fqltool':                       {'parallelism': 1, 'exec_resource_class': 'medium'},
+                'utests_long':                          {'parallelism': 1, 'exec_resource_class': 'medium'},
+                'utests_stress':                        {'parallelism': 1, 'exec_resource_class': 'medium'},
+            },
+        },
+        'HIGHER': {
+            # apply defaults to all jobs, so the below just overrides jobs which do not match the default
+            '_default_job_':                            {'parallelism': 100,    'exec_resource_class': 'xlarge'},
+            'jobs': {
+                'j8_jvm_upgrade_dtests':                {'parallelism': 2},
+                'j11_jvm_dtests':                       {'parallelism': 5},
+                'j11_jvm_dtests_vnode':                 {'parallelism': 5},
+                'j8_jvm_dtests':                        {'parallelism': 5},
+                'j8_jvm_dtests_vnode':                  {'parallelism': 5},
```

Comment:
don't know why MID/HIGh use different values here... ideally we should make this based off tests...

in my personal scripts I have the following to figure out better parallel

```
ef java_parallelism(src_dir, kind, num_file_in_worker, include = lambda a, b: True):
    d = os.path.join(src_dir, 'test', kind)
    num_files = 0
    for root, dirs, files in os.walk(d):
        for f in files:
            if f.endswith('Test.java') and include(os.path.join(root, f), f):
                num_files += 1
    return math.floor(num_files / num_file_in_worker)
...
    unit_parallelism                = java_parallelism(args.src, 'unit', 20)
    jvm_dtest_parallelism           = java_parallelism(args.src, 'distributed', 4, lambda full, name: 'upgrade' not in full)
    jvm_dtest_upgrade_parallelism   = java_parallelism(args.src, 'distributed', 2, lambda full, name: 'upgrade' in full)
```

if you use that script our jobs run faster and workflows complete faster...

Reply:
agree, just pointing out its a waste of resources; we could be nicer if we don't need xlarge